res_freeze_check.c

```diff
@@ -413,11 +414,13 @@ static int load_module(void)
 		if (everything_present) {
 			dlerror();
 			/* NOTE(afournier): maybe an issue with name mangling */
-			ast_queues_get_container = dlsym(app_queue->module->lib, "ast_queues_get_container");
-			if (!ast_queues_get_container) {
+			fn_ptr = dlsym(app_queue->module->lib, "ast_queues_get_container");
```

Comment:
Seems to work for me if you cast into a function pointer
```suggestion
			struct ao2_container* (*fn_ptr)(void) = dlsym(app_queue->module->lib, "ast_queues_get_container");
```

Reply:
I believe you have to cast the result into a pointer to avoid crashing, e.g:
`fn_ptr = (int *) dlsym(app_queue->module->lib, "ast_queues_get_container");`
would also work

Reply:
That's the weird thing! It works on my machine but not on any other :rofl:

Reply:
> I believe you have to cast the result into a pointer to avoid crashing, e.g: `fn_ptr = (void *)dlsym(app_queue->module->lib, "ast_queues_get_container");` would also work

Already tried, and all variations of it.